### PR TITLE
Update minimum php version to 7.4 and use its features

### PR DESCRIPTION
**Pull request by bronek89**
- short closures
- covariant return types
- typed properties

done using https://github.com/rectorphp/rector

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,12 +19,13 @@
...
-        "php": "^7.1"
+        "php": "^7.4"
     },
     "require-dev": {
-        "phpspec/phpspec": "^5.1",
+        "phpspec/phpspec": "^6.1",
         "phpstan/phpstan": "^0.11",
-        "phpbench/phpbench": "@dev"
+        "phpbench/phpbench": "@dev",
+        "rector/rector": "^0.6"
```
Comment (dlakomski) [inline composer.json]: I don't think we should use rector as our dev dependency. It should be perhaps used as third party software to run it once and then forgotten ;)

File: src/ArrayValue.php [hunk 2]
```diff
@@ -2,21 +2,23 @@
...
-interface ArrayValue extends Value, Collection, Stack, \IteratorAggregate, \ArrayAccess
+use IteratorAggregate;
+use ArrayAccess;
+interface ArrayValue extends Value, Collection, Stack, IteratorAggregate, ArrayAccess
 {
     // Collection
 
     /**
      * @param callable $callback function(mixed $value): void
      * @return ArrayValue
```
Comment (dlakomski) [inline src/ArrayValue.php:13]: `@return` docblocks are not needed now.

**Review by dlakomski — CHANGES_REQUESTED** [review]

File: src/ArrayValue.php (hunk 2 above)
Comment (dazet) [reply src/ArrayValue.php:13]: On the other hand it will have to be restored some day with generics templates.

Comment (bronek89) [reply src/ArrayValue.php:13]: @dazet yes, I've already started working on it: https://github.com/gowork/values/commit/dae107f33db838292d09add1e31ed3b58b2c19a2 :)

**Review by dazet — APPROVED** [review]
:+1:

File: composer.json (hunk 1 above)
Comment (bronek89) [reply composer.json]: removed :+1:

**Pull request merged**